hspec-discover/src/Run.hs

```diff
@@ -119,15 +119,31 @@ findSpecs src = do
   mapMaybe (fileToSpec dir) . filter (/= file) <$> getFilesRecursive dir
 
 fileToSpec :: FilePath -> FilePath -> Maybe Spec
-fileToSpec dir file = case reverse $ splitDirectories file of
-  x:xs -> case stripSuffix "Spec.hs" x <|> stripSuffix "Spec.lhs" x of
-    Just name | (not . null) name -> Just . Spec (dir </> file) $ (intercalate "." . reverse) (name : xs)
-    _ -> Nothing
-  _ -> Nothing
+fileToSpec dir sfile = do
+    file@(_:_) <- stripSuffix "Spec.hs" sfile <|>
```

Comment:
Checking for `null` like his works, but I'm not sure if I like it better than an explicit guard.